### PR TITLE
UCP/PROTO: Use uct_iface_estimate_perf API to query UCT performance

### PR DESCRIPTION
**Pull request by yosefe**
## Why
Have a fine-grained performance estimation per operation type

### CODE REVIEW

**Comment by yosefe** [comment]
@Artemy-Mellanox @dmitrygx can you pls review?

File: src/ucp/proto/proto_common.c [hunk 1]
```diff
@@ -146,27 +146,45 @@ static void ucp_proto_common_update_lane_perf_by_distance(
...
 {
-    const uct_iface_attr_t *iface_attr =
-            ucp_proto_common_get_iface_attr(&params->super, lane);
-    ucp_worker_h worker   = params->super.worker;
-    ucp_context_h context = worker->context;
+    ucp_worker_h worker        = params->super.worker;
+    ucp_context_h context      = worker->context;
+    ucp_rsc_index_t rsc_index  = ucp_proto_common_get_rsc_index(&params->super,
+                                                                lane);
+    ucp_worker_iface_t *wiface = ucp_worker_iface(worker, rsc_index);
```
Comment (dmitrygx) [inline src/ucp/proto/proto_common.c:158]: [minor] can it be `const`?

**Review by dmitrygx — APPROVED** [review]

**Review by brminich — APPROVED** [review]

File: src/ucp/proto/proto_common.c (hunk 1 above)
Comment (yosefe) [reply src/ucp/proto/proto_common.c:158]: in all other places, we don't use const qualifiers for objects like iface, worker, context, etc. and also ucp_worker_iface() returns a non-const pointer

Comment (dmitrygx) [reply src/ucp/proto/proto_common.c:158]: I see, thanks

**Pull request merged**